contracts/src/Assets.sol

```diff
@@ -68,6 +68,11 @@ library Assets {
             // If the final destination chain is not AssetHub, then the fee needs to additionally
             // include the cost of executing an XCM on the final destination parachain.
             costs.foreign = $.assetHubReserveTransferFee + destinationChainFee;
+
+            // Destination fee cannot be zero. MultiAssets are not allowed to be zero in xcm v4.
+            if (destinationChainFee == 0 || destinationChainFee > $.destinationMaxTransferFee) {
+                revert InvalidDestinationFee();
+            }
```

Comment:
Why not move this up above the `costs.foreign` assignment?

Reply:
addressed in [ddbdd58](https://github.com/Snowfork/snowbridge/pull/1168/commits/ddbdd58ac013440a585ed01f30a5dfad3f091e11)